### PR TITLE
feat: job and task variable add parameters description, group and advanced.

### PR DESCRIPTION
**Pull request by tao-xinxiu**

### CODE REVIEW

**Review by fviale — APPROVED** [review]
LGTM

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobVariable.java [hunk 1]
```diff
@@ -91,56 +105,70 @@ public void setModel(String model) {
...
-        return true;
+        if (getDescription() != null ? !getDescription().equals(that.getDescription()) : that.getDescription() != null)
+            return false;
+        if (getGroup() != null ? !getGroup().equals(that.getGroup()) : that.getGroup() != null)
+            return false;
+        return getAdvanced() == that.getAdvanced();
     }
 
     @Override
     public int hashCode() {
```
Comment (mbenguig) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobVariable.java:159]: Need a short comment about the hashing function

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobVariable.java [hunk 2]
```diff
@@ -91,56 +105,70 @@ public void setModel(String model) {
...
-        final int primeNumber = 31;
-        int result = 1;
-        result = primeNumber * result + ((name == null) ? 0 : name.hashCode());
-        result = primeNumber * result + ((value == null) ? 0 : value.hashCode());
-        result = primeNumber * result + ((model == null) ? 0 : model.hashCode());
+        int result = getName() != null ? getName().hashCode() : 0;
```
Comment (mbenguig) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobVariable.java:160]: Need a brief explanation about the hash function

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/TaskVariable.java [hunk 3]
```diff
@@ -68,60 +74,30 @@ public void setJobInherited(boolean inherited) {
...
-            return false;
-        }
-        return true;
+        TaskVariable that = (TaskVariable) o;
+
+        return isJobInherited() == that.isJobInherited();
     }
 
     @Override
     public int hashCode() {
```
Comment (mbenguig) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/TaskVariable.java:91]: Need a brief comment to explain the "formula"

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/TaskVariable.java [hunk 4]
```diff
@@ -68,60 +74,30 @@ public void setJobInherited(boolean inherited) {
...
-        final int primeNumber = 31;
-        int result = 1;
-        result = primeNumber * result + (jobInherited ? 3 : 5);
-        result = primeNumber * result + ((getName() == null) ? 0 : getName().hashCode());
-        result = primeNumber * result + ((getValue() == null) ? 0 : getValue().hashCode());
-        result = primeNumber * result + ((getModel() == null) ? 0 : getModel().hashCode());
+        int result = super.hashCode();
+        result = 31 * result + (isJobInherited() ? 1 : 0);
```
Comment (mbenguig) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/TaskVariable.java:93]: Need a brief comment to explain the "formula"

**Review by mbenguig — APPROVED** [review]
2 very small comments

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobVariable.java (hunk 2 above)
Comment (fviale) [reply scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/JobVariable.java:160]: this is an auto-generated hashcode method

File: scheduler/scheduler-api/src/main/resources/org/ow2/proactive/scheduler/common/xml/schemas/jobdescriptor/dev/schedulerjob.rnc [hunk 5]
```diff
@@ -30,13 +30,19 @@ jobName = ## Identification of this job
 variables = ## Definition of variables which can be reused throughout this descriptor
     element variables { variable+ }
 variable = ## Definition of one variable, the variable can be reused (even in another following variable definition) by using the syntax ${name_of_variable}
-    element variable { variableName, variableValue, variableModel?}
+    element variable { variableName, variableValue, variableModel?, variableDescription?, variableGroup?, variableAdvanced? }
```
Comment (fviale) [inline scheduler/scheduler-api/src/main/resources/org/ow2/proactive/scheduler/common/xml/schemas/jobdescriptor/dev/schedulerjob.rnc:33]: I forgot to mention it, but could you please add these changes in schemas/globalvariables

(you will need to create a new folder 3.13, and also change the schema version in the config/scheduler/global_variables.xml file)

**Review by fviale — APPROVED** [review]
LGTM

**Pull request merged**